### DISCUSSION TITLE
Account Switch does not persist in the app for WalletConnectV2,the app displays the address of the previously connected account instead of Current One

### DISCUSSION THREAD

**Discussion by arunkumar201** [WalletConnect/walletconnect-monorepo#3423] [Developer Support]
When manually switching to a new account using the Account Switch feature in the mobile wallet, the app immediately updates the UI to show the details of the new account, including the account address and balance. However, if the page is reloaded or refreshed, the app does not retain the state of the Account Switch feature. Instead, it displays the address of the previously connected account. This issue was tested on the official website https://cloud.walletconnect.com/sign-in.
**Reproduction Steps**
1. Visit https://cloud.walletconnect.com/sign-in
2. Connect with MetaMask (the latest version for mobile)
3. Switch the account
4. Refresh the page, and the previously connected account will be displayed instead of the currently connected account.

**Screenshots**

**Comment by glitch-txs** [comment]
Hi, have you tested this with other wallet? There's currently some issues with chain switching on MetaMask.

**Comment by arunkumar201** [comment]
This issue does not occur when we connect using the MetaMask extension. It only occurs when we use the MetaMask wallet. I tried testing with Trust Wallet, but I am unable to connect on https://cloud.walletconnect.com/sign-in. When I click on the connect button in Trust Wallet after scanning, sometimes the connect button does not open. I have reinstalled and tried, but the same thing is working fine with the my demo that I prepared for Trust Wallet.

**Comment by glitch-txs** [comment]
So this only happens on Cloud website? Just trying to target the issue

**Comment by arunkumar201** [comment]
No. I am encountering the main issue ##3327 in my demo, project, and Cloud website as well just took the Cloud website to see ad verified the bug.

**Comment by glitch-txs** [comment]
MetaMask issue is being worked on.

 Could you try labs with Trust Wallet? Is it android or iOS? Thanks.

https://lab.web3modal.com/

**Reply by ccummings** [reply]
What's the issue with Metamask? I'm encountering a problem using MM and switching accounts not working correctly.

**Comment by arunkumar201** [comment]
Android OS, I am not able to perform the account switch on Trust Wallet it is my Fault.
